Library/Homebrew/os/linux/elf.rb

```diff
@@ -0,0 +1,125 @@
+module ELFShim
+  def elf?
+    return @elf if defined? @elf
+    # See: https://en.wikipedia.org/wiki/Executable_and_Linkable_Format#File_header
+    return @elf = false unless read(4) == "\x7fELF"
```

Comment:
I'd say pretty much every number in this file should be a constant explaining what it means otherwise they are a bit "magic".